src/middleware.ts

```diff
@@ -0,0 +1,26 @@
+import { type NextRequest, NextResponse } from "next/server";
+import { getSession } from "../utils/supabase/getSession";
+import { Routes } from "./lib/routes";
+
+export default async function updateSession(request: NextRequest) {
+  try {
+    const isTokenValid = await getSession();
+    const requestedPath = request.nextUrl.pathname;
+
+    if (!isTokenValid && requestedPath !== Routes.Login) {
+      return NextResponse.redirect(new URL(Routes.Login, request.url));
+    }
+
+    if (requestedPath === "/") {
```

Comment:
Can you explain it why we need it ?
 i think / is equal to main route not indicate all routes